src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesTreeRenderer.java

```diff
@@ -65,8 +65,11 @@ public void customizeCellRenderer(@NotNull JTree tree,
         append(" [" + exerciseViewModel.getStatusText() + "]", STATUS_TEXT_STYLE);
       }
       setEnabled(exerciseViewModel.isSubmittable());
+      String submittable = ExerciseViewModel.Status.TUTORIAL.equals(exerciseViewModel.getStatus())
+          ? getText("ui.exercise.ExercisesTreeRenderer.useStartButtonTutorial")
+          : getText("ui.exercise.ExercisesTreeRenderer.useUploadButton");
       setToolTipText(exerciseViewModel.isSubmittable()
-          ? getText("ui.exercise.ExercisesTreeRenderer.useUploadButton")
+          ? submittable
```

Comment:
Inlining this variable would avoid calculating its value for non-submittable exercises.

Reply:
At first, I had that as an inline expression, but Sonnar complained.